packages/protocol/contracts/psp/AirnodeSponsorPspAllocator.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "../rrp/interfaces/IAirnodeRrp.sol";
+import "./AirnodePsp.sol";
+
+// It would be nice if slots could also expire, similar to whitelisting expiration
+contract AirnodeSponsorPspAllocator {
+    struct Slot {
+        bytes32 subscriptionId;
+        address sponsor;
+    }
+
+    IAirnodeRrp public airnodeRrp;
+    AirnodePsp public airnodePsp;
+    mapping(address => mapping(address => uint256))
+        public airnodeToSponsorToNoOfTotalSlots;
+    mapping(address => mapping(address => uint256))
+        public airnodeToSponsorToNoOfUsedSlots;
+    mapping(address => mapping(uint256 => Slot))
+        public airnodeToSlotIndexToSlot;
+
+    constructor(address airnodeRrp_, address airnodePsp_) {
+        airnodeRrp = IAirnodeRrp(airnodeRrp_);
+        airnodePsp = AirnodePsp(airnodePsp_);
+    }
+
+    function setTotalNoOfSlots(
```

Comment:
What is the use case for this?

Reply:
You allocate N slots to a sponsor (because they paid you or you signed an agreement), so they can make N consecutive subscriptions but not any more (because slots are a finite resource)

Reply:
Actually I misread this function. This sets the number of slots for a particular sponsor (I thought it's global).

Anyway, shouldn't the contract support setting a global maximum number of slots? Since, this number might impact the Airnode throughtput and I also see a business value here (provably saying that a given allocator only serves at most X subscribers)

Reply:
A global maximum number of slots can only be set at the node side (in the config file) because the allocator contracts don't know about each other. When the config file uses one allocator contract for 5 slots and another for 10, it implicitly has a limit of 15 on the global number of slots.